tests/unit/swagger/templateHelpers.spec.ts

```diff
@@ -264,6 +264,76 @@ describe('ValidationService', () => {
     });
   });
 
+  describe('Enum validate', () => {
+    type Enumeration = string[] | number[];
+
+    it('should enum number value', () => {
+      const name = 'name';
+      const value = '1';
+      const error = {};
+      const enumeration: Enumeration = ['0', '1'];
+      const result = new ValidationService({}).validateEnum(name, value, error, enumeration);
+      expect(result).to.equal(value);
+    });
+
+    it('should enum string value', () => {
+      const name = 'name';
+      const value = 'HELLO';
+      const error = {};
+      const enumeration: Enumeration = ['HELLO'];
+      const result = new ValidationService({}).validateEnum(name, value, error, enumeration);
+      expect(result).to.equal(value);
+    });
+
+    it('should enum no member', () => {
+      const name = 'name';
+      const value = 'HI';
+      const error: any = {};
+      const enumeration: Enumeration = [];
+      const result = new ValidationService({}).validateEnum(name, value, error, enumeration);
+      expect(result).to.equal(undefined);
+      expect(error[name].message).to.equal(`no member`);
+    });
+
+    it('should enum out of member', () => {
+      const name = 'name';
+      const value = 'SAY';
+      const error: any = {};
+      const enumeration: Enumeration = ['HELLO', 'HI'];
+      const result = new ValidationService({}).validateEnum(name, value, error, enumeration);
+      expect(result).to.equal(undefined);
+      expect(error[name].message).to.equal(`should be one of the following; ['HELLO', 'HI']`);
+    });
+
+    it('accepts a string value of a numeric enum', () => {
+      const name = 'name';
+      const value = '1';
+      const error: any = {};
+      const enumeration: Enumeration = [0, 1];
+      const result = new ValidationService({}).validateEnum(name, value, error, enumeration);
+      expect(result).to.equal(value);
+    });
+
+    it('accepts a numeric value of a string-numeric enum', () => {
```

Comment:
Same for this one as above (https://github.com/lukeautry/tsoa/pull/332/files/fa05ae3c9aeefa741c2643909b5b07655b0fe9ed#discussion_r327400448). This case should be a negative test case. We can’t allow a number to swap as a string.